app/lib/repository.rb

```diff
@@ -254,7 +254,9 @@ def self.get_all_permissions
       end
       # NOTE: this will allow graders to access the files in the entire repository
```

Comment:
Unrelated to this PR but we should deal with this note eventually. Especially in this context where one assignment could have anonymization enabled and the other has it disabled.